src/libasr/pass/replace_symbolic.cpp

```diff
@@ -0,0 +1,175 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/replace_symbolic.h>
+#include <libasr/pass/pass_utils.h>
+
+#include <vector>
+
+
+namespace LCompilers {
+
+using ASR::down_cast;
+using ASR::is_a;
+
+
+class ReplaceSymbolicVisitor : public PassUtils::PassVisitor<ReplaceSymbolicVisitor>
+{
+public:
+    ReplaceSymbolicVisitor(Allocator &al_) :
+    PassVisitor(al_, nullptr) {
```

Comment:
If this: https://github.com/lcompilers/lpython/pull/2200#discussion_r1274403372 doesn't work, pass `m_global_scope` as an argument to `ReplaceSymbolicVisitor` and pass that `symtab` to `PassVisitor`.